Problem4/src/problem4/Ball.java

```diff
@@ -0,0 +1,68 @@
+/*
+ * To change this license header, choose License Headers in Project Properties.
+ * To change this template file, choose Tools | Templates
+ * and open the template in the editor.
+ */
+package problem4;
+
+import java.util.Random;
+
+/**
+ *
+ * @author manxboy
+ */
+public class Ball {
+    public Vec2d pos;
+    public Vec2d velocity;
+    
+    public Ball(Vec2d pos) {
+        this.pos = pos;
+        this.velocity = Vec2d.RIGHT;
+    }
+    
+    
+    /**
+     * Test wether this ball will collide with the other ball in this frame
+     * @param b the other ball to test ageist
+     * @return true if colliding/will collide, else false
+     */
+    public boolean willCollideWithBallThisFrame(Ball b) {
```

Comment:
Since this method is only called from within the Ball class, it can be made private.